x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/page.tsx

```diff
@@ -51,6 +54,20 @@ export const Page: FC<{
     checkJobsExist();
   }, []);
 
+  useEffect(
+    function updateUrl() {
+      if (analyticsId !== undefined) {
+        setGlobalState({
+          ml: {
+            ...(analyticsId.analysis_type ? { analysisType: analyticsId.analysis_type } : {}),
+            ...(analyticsId.job_id ? { jobId: analyticsId.job_id } : {}),
+          },
+        });
+      }
+    },
+    [analyticsId?.job_id, analyticsId?.model_id]
+  );
```

Comment:
shall we store it in the `_a` state instead? the `_g` is intended to be used as a global Kibana URL state

Reply:
I reckon we don't need an intermediate `analyticsId` state here. The value should be provided directly from the URL and updated as well

Reply:
In this case the value wouldn't be in the url - we're adding it into the url when the user selects the analytics id from the selection flyout. Moving it to `_a` would require some refactoring so I think it's outside the scope of this for now and better suited as an improvements or maintenance task.

Reply:
The `_g` parameter was originally intended for any settings which might want to be preserved across pages. This is why we put the job ID for the anomaly detection pages in `_g`, so that on switching from the Anomaly Explorer to the Single Metric Viewer for example, the selected job would be retained. So using `_g` would make sense here too, so that in theory the job would be retained if the user switched from the Results view to the Map view. Whether using `_g` and `_a` in the long term still makes sense is another matter entirely, but here makes total sense to use `_g` for analytics job ID.